pkg/gadgets/snapshot/process/types/process-collector_test.go

```diff
@@ -0,0 +1,196 @@
+// Copyright 2022 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package types
+
+import (
+	"bufio"
+	"fmt"
+	"strings"
+	"testing"
+)
+
+type stsTestCase struct {
+	val      uint64
+	valMax   uint64
+	width    uint64
+	expected string
+}
+
+func TestCreateTree(t *testing.T) {
```

Comment:
nit: An alternative could be to test the whole created tree. 

```golang
func TestCreateTree(t *testing.T) {
	t.Parallel()

	pid42 := Event{
		Pid: 42,
	}

	pid43 := Event{
		Pid:       43,
		ParentPid: 42,
	}

	pid44 := Event{
		Pid:       44,
		ParentPid: 42,
	}

	pid45 := Event{
		Pid:       45,
		ParentPid: 42,
	}

	pid46 := Event{
		Pid:       46,
		ParentPid: 43,
	}

	events := []*Event{
		&pid42,
		&pid43,
		&pid44,
		&pid45,
		&pid46,
	}

	tree, err := createTree(events)
	if err != nil {
		t.Fatalf("fail to create tree: %v", err)
	}

	expectedTree := &processTree{
		process: &pid42,
		children: []*processTree{
			{
				process: &pid43,
				children: []*processTree{
					{
						process:  &pid46,
						children: []*processTree{},
					},
				},
			},
			{
				process:  &pid44,
				children: []*processTree{},
			},
			{
				process:  &pid45,
				children: []*processTree{},
			},
		},
	}

	require.Equal(t, expectedTree, tree, "tree is wrong")
}
```

The disadvantage is that the provided information is less verbose, but IMO the test is much more readable.